internal/pkg/apply/apply_test.go

```diff
@@ -59,3 +66,321 @@ func TestApply(t *testing.T) {
+
+	// set up a kustomization
+	kp := wiretest.InitializConfigProvider()
+	fs, cleanup, err := InitializeKustomizationWithPresence()
+	defer cleanup()
+	assert.NoError(t, err)
+	assert.Equal(t, len(fs), 2)
+
+	// confirm the loaded resources containing a service
+	// without the annotation for EnsureDoesNotExist
+	objects, err := kp.GetConfig(fs[0])
+	assert.NoError(t, err)
+	service := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "v1",
+			"kind":       "Service",
+			"metadata": map[string]interface{}{
+				"name":      "my-service",
+				"namespace": "default",
+			},
+			"spec": map[string]interface{}{
+				"ports": []interface{}{
+					map[string]interface{}{
+						"port":       int64(80),
+						"protocol":   "TCP",
+						"targetPort": int64(9376)},
+				},
+				"selector": map[string]interface{}{"app": "MyApp"},
+			},
+		},
+	}
+	assert.NoError(t, err)
+	assert.Contains(t, objects, service)
+
+	// Initialize the Apply object
+	a, done, err := wiretest.InitializeApply(objects, &object.Commit{}, buf)
+	defer done()
+	assert.NoError(t, err)
+
+	// Apply the first kustomization
+	// Confirm there is one Service object
+	// Confirm the Service object is the expected one
+	r, err := a.Do()
+	assert.NoError(t, err)
+	assert.Equal(t, apply.Result{objects}, r)
+	liveService := service.DeepCopy()
+	exist, err := util.ObjectExist(a.DynamicClient, context.Background(), liveService)
+	assert.NoError(t, err)
+	assert.Equal(t, true, exist)
+	assert.Equal(t, false, util.HasAnnotation(liveService, constants.Presence))
+
+	// Update the kustomization
+	// by adding the annotation EnsureDoesNotExist to the Service object
+	// Confirm there is a Service resource with the annotation EnsureDoesNotExist
+	updatedObjects, err := kp.GetConfig(fs[1])
+	assert.NoError(t, err)
+	updatedService := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "v1",
+			"kind":       "Service",
+			"metadata": map[string]interface{}{
+				"name":      "my-service",
+				"namespace": "default",
+				"annotations": map[string]interface{}{
+					"kubectl.kubernetes.io/presence": "EnsureDoesNotExist",
+				},
+			},
+			"spec": map[string]interface{}{
+				"ports": []interface{}{
+					map[string]interface{}{
+						"port":       int64(80),
+						"protocol":   "TCP",
+						"targetPort": int64(9376)},
+				},
+				"selector": map[string]interface{}{"app": "MyApp"},
+			},
+		},
+	}
+	assert.Contains(t, updatedObjects, updatedService)
+	assert.NotEqual(t, service, updatedService)
+
+	// Apply the second kustomization
+	// Confirm that the Service object is not updated
+	a.Resources = updatedObjects
+	r, err = a.Do()
+	assert.NoError(t, err)
+	assert.Equal(t, apply.Result{updatedObjects}, r)
+	exist, err = util.ObjectExist(a.DynamicClient, context.Background(), liveService)
+	assert.Equal(t, true, exist)
+	assert.NoError(t, err)
+	assert.Equal(t, false, util.HasAnnotation(liveService, constants.Presence))
```

Comment:
can you also manually delete the live service, re-apply and verify it is not re-created

Reply:
done